### PR TITLE
JMS Alarm Northbounder

### PR DESCRIPTION
**Pull request by dschlenk**
Also cleans up formatting in some parts of the syslog alarm northbounder and does a bit of refactoring to improve code reuse.

### CODE REVIEW

**Comment by soleger** [comment]
The only suggestion that I have is that you can probably replace your TestNodeDao by using the MockNodeDao which is available in applicationContext-mockDao.xml.

**Comment by dschlenk** [comment]
@soleger I'll take a look and rip out the syslog NBI's as well.

**Comment by pbrane** [comment]
> On Jul 16, 2015, at 3:36 PM, David Schlenk <notifications@github.com> wrote:
> 
> @soleger <https://github.com/soleger> I'll take a look and rip out the syslog NBI's as well.
> 
Thanks, David.

File: opennms-alarms/api/src/main/java/org/opennms/netmgt/alarmd/api/support/AbstractNorthbounder.java [hunk 1]
```diff
@@ -147,21 +171,149 @@ public void run() {
                         Thread.sleep(m_retryInterval);
                     }
                 }
-            
+
             }
-        
+
         } catch (InterruptedException e) {
             // thread interrupted so complete it
```
Comment (mvrueden) [inline opennms-alarms/api/src/main/java/org/opennms/netmgt/alarmd/api/support/AbstractNorthbounder.java]: How about a LOG.warn message here?

`LOG.warn("Thread '{}' was interrupted unexpected.", getName());

File: opennms-alarms/api/src/main/java/org/opennms/netmgt/alarmd/api/support/AbstractNorthbounder.java [hunk 2]
```diff
@@ -147,21 +171,149 @@ public void run() {
...
+        mapping.put("x733AlarmType", alarm.getX733Type());
+
+        try {
+            mapping.put("x733ProbableCause",
+                        nullSafeToString(x733ProbableCause.get(alarm.getX733Cause()),
+                                         ""));
+        } catch (Exception e) {
+            LOG.info("Exception caught setting X733 Cause: {}",
+                     alarm.getX733Cause(), e);
+            mapping.put("x733ProbableCause", "");
```
Comment (mvrueden) [inline opennms-alarms/api/src/main/java/org/opennms/netmgt/alarmd/api/support/AbstractNorthbounder.java]: If we were not able to map this, shouldn't we set something like "ERROR" or "UNKNOWN"?

**Comment by dschlenk** [comment]
Trying to get it to work, following what opennms-provision/opennms-provisiond's ITs do I added `"classpath:/META-INF/opennms/applicationContext-mockDao.xml"` to `@ContextConfiguration`'s locations list and `org.opennms:opennms-dao`, `org.opennms:opennms-dao-mock` to the maven test scope, but running the tests fails to load the application context because:
```
Caused by: java.io.FileNotFoundException: ${opennms.home}/etc/model-importer.properties (No such file or directory)
```
(from opennms-dao-mock/src/main/resources/META-INF/opennms/applicationContext-mockDao.xml. )

I also removed the PropertyPlaceholderConfigurer stuff from my test context since there seemed to be some conflicts there (and I didn't really need it). 

@soleger am I missing another dependency or location in `@ContextConfiguration`? 

Pushed my attempt to [dschlenk/jms_nb_mockNodeDao](https://github.com/dschlenk/opennms/tree/jms_nb_mockNodeDao)

File: opennms-alarms/api/src/main/java/org/opennms/netmgt/alarmd/api/support/AbstractNorthbounder.java (hunk 2 above)
Comment (dschlenk) [inline opennms-alarms/api/src/main/java/org/opennms/netmgt/alarmd/api/support/AbstractNorthbounder.java]: Probably - I just moved this method up from the SyslogNorthbounder so I could reuse it.

**Comment by dschlenk** [comment]
Silly me, forgot to include the `@JUnitConfigurationEnvironment` annotation on my test class.

**Comment by dschlenk** [comment]
Anything else I need to do to get this merged?

**Comment by soleger** [comment]
I think we just need to run it through bamboo and then we can merge it. I'll work on that as soon as I can. I looked at the MockNodeDao stuff and it looks good. Thanks!

**Comment by dschlenk** [comment]
No worries @soleger just wanted to make sure I wasn't holding anything up.

**Comment by soleger** [comment]
Just added a branch for this in bamboo, after it passes I'll go ahead and merge.

**Comment by soleger** [comment]
Just merged this in for inclusion in 17. Thanks for the great work David!

**Pull request merged**